src/datastore/offline-dummy-store.ts

```diff
@@ -0,0 +1,284 @@
+import { EventEmitter } from 'events';
```

Comment:
This is a lot of code that requires maintaining as the `DataStore` interface evolves. Something like `db = {} as DataStore;` would have the same effect.

Alternatively, a js `proxy` could be used to throw a helpful error if a function is accidentally called. Something like:
```ts
export const OfflineDummyStore: DataStore = new Proxy(new EventEmitter() as DataStore, {
  get(target: any, propKey) {
    if (propKey in target) return target[propKey];
    return function () {
      throw new Error(
        `Cannot call function on the Dummy datastore. Check if the application is running in offline mode.`
      );
    };
  },
});
```